components/net/websocket_loader.rs

```diff
@@ -331,8 +324,34 @@ async fn start_websocket(
     let socket = try_socket.map_err(Error::Io)?;
     let connector = TlsConnector::from(Arc::new(tls_config));
 
+    let mut url = client.url();
+    if url.scheme() == "https" {
+        url.as_mut_url()
+            .set_scheme("wss")
+            .expect("Can't set scheme from https to wss");
+    } else {
+        url.as_mut_url()
+            .set_scheme("ws")
+            .expect("Can't set scheme from http to ws");
+    };
+    // TODO(pylbrecht): move request conversion to a separate function
+    let mut builder = ClientRequestBuilder::new(
+        url.clone() // FIXME(pylbrecht): deal with clone()
+            .into_string()
+            .parse()
+            .expect("unable to parse URI"),
+    );
+    for (key, value) in client.headers.iter() {
+        builder = builder.with_header(
+            key.as_str(),
+            value
+                .to_str()
+                .expect("unable to convert header value to string"),
+        );
+    }
```

Comment:
I thought about implementing [`IntoClientRequest`](https://docs.rs/tungstenite/0.27.0/tungstenite/client/trait.IntoClientRequest.html) for `net_traits::request::Request`, but I think that would require adding `aysync-tungstenite` to the `net_traits` crate. Though adding a dependency just for the sake of having a cleaner conversion is probably not worth it. A simple function performing the conversion might suffice here.